graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/BaseDgsQueryExecutor.kt

```diff
@@ -108,23 +109,22 @@ object BaseDgsQueryExecutor {
         idProvider.ifPresent { graphQLBuilder.executionIdProvider(it) }
 
         val graphQL: GraphQL = graphQLBuilder.build()
-
-        val dataLoaderRegistry = dataLoaderProvider.buildRegistryWithContextSupplier { dgsContext }
+        val graphQLContextFuture = CompletableFuture<GraphQLContext>()
+        val dataLoaderRegistry = dataLoaderProvider.buildRegistryWithContextSupplier { graphQLContextFuture.get() }
```

Comment:
The changes look fine. I'm more concerned about the breaking change, and we do seem to jave quite a few usages that assume the type of context available in the `BatchLoaderEnvironment` is a `DgsContext`. One potential way to avoid breaking users, is to check whether `GraphQlContextContributors` is set in the BaseDgsQueryExecutor, and setting the context in the DataLoaderProvider accordingly. Then existing users will still get a DgsContext and the folks using the newer GraphQLContext will get that instead.

This assumes the user is choosing between either the `DgsContext` mechanism or using the newer `GraphQLContextContributor` mechanism to set up the custom context and not a mix of both approaches.

We can then get rid/deprecate the DgsContext in a future breaking release. Thoughts?

Reply:
That's a good idea. That should significantly reduce the chance of breaking existing code.

Reply:
Or what about conversely checking whether dgsContext.customContext is set?

Reply:
That would work, assuming it is not always set up with some default.

Reply:
Actually, not. Folks may be accessing the dgs context in the batch loader environment even if they aren't setting up a custom context,

Reply:
We discussed this PR offline and concluded we will make this a breaking change in DGS `6.0.0` release along with Spring Boot 3.0 update instead of trying to make it backwards compatible.